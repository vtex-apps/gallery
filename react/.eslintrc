{
  "extends": "vtex-react",
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  },
  "rules": {
    "react/jsx-indent": "off"
  }
}
