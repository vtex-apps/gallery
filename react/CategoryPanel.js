import React, { Fragment } from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { Link } from 'vtex.render-runtime'
import { facetOptionShape } from './constants/propTypes'
import searchResult from './searchResult.css'
import CategoriesHighlights from 'vtex.store-components/CategoriesHighlights'

const renderCategoryShelf = (category, noOfChildren, quantityOfItemsPerRow) => {
  const categoriesHighlighted = category.children
    .slice(0, noOfChildren)
    .map(child => ({
      id: child.id,
      name: child.name,
      to: child.link,
    }))
  const headerClasses = classNames(
    searchResult.categoryPanelHeaderRow,
    'flex flex-row flex-wrap items-stretch pa3 bn mh4-ns'
  )
  const itemClasses = classNames(
    searchResult.categoryPanelItemRow,
    'flex flex-row flex-wrap items-stretch pa3 bn mh4-ns'
  )
  let i = 0
  const items = []
  for (i = 0; i < categoriesHighlighted.length; i += quantityOfItemsPerRow) {
    items.push(categoriesHighlighted.slice(i, i + quantityOfItemsPerRow))
  }
  return (
    <Fragment key={`parent-fragment-${category.id}`}>
      <h3 className={`t-heading-3 ${headerClasses}`}>
        <Link to={category.link}>{category.name}</Link>
      </h3>
      {items.length > 0 &&
        items.map((item, index) => (
          <div key={`category-row-${index}`} className={itemClasses}>
            <CategoriesHighlights
              categoriesHighlighted={item}
              showCategoriesHighlighted
              quantityOfItems={quantityOfItemsPerRow}
            />
          </div>
        ))}
      {category.children.map(child => {
        if (child.children && child.children.length > 0) {
          return renderCategoryShelf(child, noOfChildren, quantityOfItemsPerRow)
        }
      })}
    </Fragment>
  )
}

const CategoryPanel = ({ tree, noOfCategories, quantityOfItemsPerRow }) => {
  return (
    <div>
      {tree.map(category => {
        return renderCategoryShelf(
          category,
          noOfCategories,
          quantityOfItemsPerRow
        )
      })}
    </div>
  )
}

CategoryPanel.propTypes = {
  /** Category tree */
  tree: PropTypes.arrayOf(facetOptionShape),
  /** Number of category filters to be shown */
  noOfCategories: PropTypes.number,
  /** Number of categories per row */
  quantityOfItemsPerRow: PropTypes.number,
}

CategoryPanel.defaultProps = {
  tree: [],
  noOfCategories: 1000,
  quantityOfItemsPerRow: 4,
}

export default CategoryPanel
