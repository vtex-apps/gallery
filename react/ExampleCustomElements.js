import React, { Component } from 'react'
import SearchResults from './SearchResult'

const GalleryItem=(props) =><div>Im a card </div>
const Gallery=(props) =><div>Im a Gallery </div>
const SelectedFilters=(props) =><div>Im Selected Filters</div>
const SearchFilter=(props) =><div>Im a Filter </div>
const Spinner=(props) =><div>Loading!!! </div>

export default (props)=>{
    return <SearchResults {...props} 
        //GalleryItem={GalleryItem} 
        //Gallery={Gallery}
        //SearchFilter={SearchFilter}
        Spinner={Spinner}
        //SelectedFilters={SelectedFilters}
    />
}