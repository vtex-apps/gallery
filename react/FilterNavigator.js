import React from 'react'
import PropTypes from 'prop-types'
import { useRuntime } from 'vtex.render-runtime'
import { flatten, path, identity, contains } from 'ramda'
import ContentLoader from 'react-content-loader'
import classNames from 'classnames'
import { FormattedMessage } from 'react-intl'

import FilterSidebar from './components/FilterSidebar'
import SelectedFilters from './components/SelectedFilters'
import AvailableFilters from './components/AvailableFilters'
import {
  formatCategoriesTree,
  mountOptions,
  getMapByType,
} from './constants/SearchHelpers'
import {
  facetOptionShape,
  paramShape,
  hiddenFacetsSchema,
} from './constants/propTypes'

import searchResult from './searchResult.css'

export const CATEGORIES_TYPE = 'Categories'
export const BRANDS_TYPE = 'Brands'
export const PRICE_RANGES_TYPE = 'PriceRanges'
export const SPECIFICATION_FILTERS_TYPE = 'SpecificationFilters'

const CATEGORIES_TITLE = 'search.filter.title.categories'
const BRANDS_TITLE = 'search.filter.title.brands'
const PRICE_RANGES_TITLE = 'search.filter.title.price-ranges'

/**
 * Wrapper around the filters (selected and available) as well
 * as the popup filters that appear on mobile devices
 */
const FilterNavigator = ({
  query,
  map,
  priceRange,
  getLinkProps,
  tree = [],
  specificationFilters = [],
  priceRanges = [],
  brands = [],
  loading = false,
  hiddenFacets = {},
}) => {
  const {
    hints: { mobile },
  } = useRuntime()

  const getCategories = () => {
    if (!tree || !tree.length) {
      return []
    }
    return formatCategoriesTree(tree)
  }

  const getAvailableCategories = (showOnlySelected = false) => {
    let queryParams = query || ''

    const mapArray = map.split(',')

    const categoriesCount = mapArray.filter(
      m => m === getMapByType(CATEGORIES_TYPE)
    ).length

    const currentPath = queryParams
      .split('/')
      .filter((_, index) => mapArray[index] === getMapByType(CATEGORIES_TYPE))
      .join('/')

    return getCategories()
      .filter(c =>
        showOnlySelected
          ? c.level === categoriesCount - 1
          : c.level === categoriesCount - 1 || c.level === categoriesCount
      )
      .filter(c => c.path.toLowerCase().startsWith(currentPath.toLowerCase()))
  }

  const getSelectedFilters = () => {
    const availableCategories = getAvailableCategories(true)

    const options = [
      ...mountOptions(availableCategories, CATEGORIES_TYPE, map),
      ...specificationFilters.map(spec =>
        mountOptions(spec.facets, SPECIFICATION_FILTERS_TYPE, map)
      ),
      ...mountOptions(brands, BRANDS_TYPE, map),
      ...mountOptions(priceRanges, PRICE_RANGES_TYPE, map),
    ]

    return flatten(options).filter(opt => opt.selected)
  }

  const getFilters = () => {
    const categories = getAvailableCategories()

    const hiddenFacetsNames = (
      path(['specificationFilters', 'hiddenFilters'], hiddenFacets) || []
    ).map(filter => filter.name)

    const mappedSpecificationFilters = !path(
      ['specificationFilters', 'hideAll'],
      hiddenFacets
    )
      ? specificationFilters
          .filter(spec => !contains(spec.name, hiddenFacetsNames))
          .map(spec => ({
            type: SPECIFICATION_FILTERS_TYPE,
            title: spec.name,
            options: spec.facets,
          }))
      : []

    return [
      !hiddenFacets.categories && {
        type: CATEGORIES_TYPE,
        title: CATEGORIES_TITLE,
        options: categories,
        oneSelectedCollapse: true,
      },
      ...mappedSpecificationFilters,
      !hiddenFacets.brands && {
        type: BRANDS_TYPE,
        title: BRANDS_TITLE,
        options: brands,
      },
      !hiddenFacets.priceRange && {
        type: PRICE_RANGES_TYPE,
        title: PRICE_RANGES_TITLE,
        options: priceRanges,
      },
    ].filter(identity)
  }

  const filterClasses = classNames({
    'flex justify-center flex-auto ': mobile,
  })

  if (!map || !map.length) {
    return null
  }

  if (loading && !mobile) {
    return (
      <div className={searchResult.filters}>
        <ContentLoader
          style={{
            width: '100%',
            height: '100%',
          }}
          width="230"
          height="320"
          y="0"
          x="0"
        >
          <rect width="100%" height="1em" />
          <rect width="100%" height="8em" y="1.5em" />
          <rect width="100%" height="1em" y="10.5em" />
          <rect width="100%" height="8em" y="12em" />
        </ContentLoader>
      </div>
    )
  }

  if (mobile) {
    return (
      <div className={searchResult.filters}>
        <div className={filterClasses}>
          <FilterSidebar
            filters={getFilters()}
            selectedFilters={getSelectedFilters()}
            getLinkProps={getLinkProps}
            map={map}
          />
        </div>
      </div>
    )
  }

  return (
    <div className={searchResult.filters}>
      <div className={filterClasses}>
        <div className="bb b--muted-4">
          <h5 className="t-heading-5 mv5">
            <FormattedMessage id="search-result.filter-button.title" />
          </h5>
        </div>
        <SelectedFilters
          filters={getSelectedFilters()}
          getLinkProps={getLinkProps}
        />
        <AvailableFilters
          filters={getFilters()}
          map={map}
          priceRange={priceRange}
        />
      </div>
    </div>
  )
}

FilterNavigator.propTypes = {
  /** Get the props to pass to render's Link */
  getLinkProps: PropTypes.func.isRequired,
  /** Categories tree */
  tree: PropTypes.arrayOf(facetOptionShape),
  /** Params from pages */
  params: paramShape,
  /** List of brand filters (e.g. Samsung) */
  brands: PropTypes.arrayOf(facetOptionShape),
  /** List of specification filters (e.g. Android 7.0) */
  specificationFilters: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      facets: PropTypes.arrayOf(facetOptionShape),
    })
  ),
  /** List of price ranges filters (e.g. from-0-to-100) */
  priceRanges: PropTypes.arrayOf(facetOptionShape),
  /** Current price range filter query parameter */
  priceRange: PropTypes.string,
  /** Map query parameter */
  map: PropTypes.string,
  /** Loading indicator */
  loading: PropTypes.bool,
  ...hiddenFacetsSchema,
}

export default FilterNavigator
