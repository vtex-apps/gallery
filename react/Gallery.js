import React, { useMemo, useContext } from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { pluck, splitEvery } from 'ramda'

import { useDevice } from 'vtex.device-detector'
import { ProductListContext } from 'vtex.product-list-context'
import { useResponsiveValue } from 'vtex.responsive-values'
import { useCssHandles } from 'vtex.css-handles'
import { useRuntime } from 'vtex.render-runtime'

import { LAYOUT_MODE } from './components/LayoutModeSwitcher'
import { productShape } from './constants/propTypes'
import withResizeDetector from './components/withResizeDetector'
import GalleryRow from './components/GalleryRow'
import ProductListEventCaller from './utils/ProductListEventCaller'
import SettingsContext from './components/SettingsContext'

import { Spinner } from 'vtex.styleguide'

/** Layout with one column */
const ONE_COLUMN_LAYOUT = 1

const LAZY_RENDER_THRESHOLD = 2

const CSS_HANDLES = ['gallery']

const { ProductListProvider } = ProductListContext
/**
 * Canonical gallery that displays a list of given products.
 */
const Gallery = ({
  products = [],
  mobileLayoutMode = 'normal',
  maxItemsPerRow = {
    desktop: 5,
    tablet: 3,
    phone: 2,
  },
  minItemWidth = 240,
  width,
  summary,
  showingFacets,
  lazyItemsRemaining,
}) => {
  const { isMobile } = useDevice()
  const { trackingId = 'Search result' } = useContext(SettingsContext) || {}
  const handles = useCssHandles(CSS_HANDLES)
  const { getSettings } = useRuntime()
  const responsiveMaxItemsPerRow = useResponsiveValue(maxItemsPerRow)

  const layoutMode = isMobile ? mobileLayoutMode : 'normal'

  const getItemsPerRow = () => {
    const maxItemsPerMinWidth = Math.floor(width / minItemWidth)

    if (isMobile) {
      if (layoutMode === 'normal') {
        return ONE_COLUMN_LAYOUT
      }

      const maxItemsOnMobile =
        maxItemsPerMinWidth >= 2 ? maxItemsPerMinWidth : 2

      return responsiveMaxItemsPerRow <= maxItemsOnMobile
        ? responsiveMaxItemsPerRow
        : maxItemsOnMobile
    }

    return responsiveMaxItemsPerRow <= maxItemsPerMinWidth
      ? responsiveMaxItemsPerRow
      : maxItemsPerMinWidth
  }

  const itemsPerRow = getItemsPerRow() || responsiveMaxItemsPerRow

  const rows = useMemo(() => splitEvery(itemsPerRow, products), [
    itemsPerRow,
    products,
  ])

  const galleryClasses = classNames(
    handles.gallery,
    'flex flex-row flex-wrap items-stretch bn ph1 na4',
    {
      'justify-center': !showingFacets,
      'pl9-l': showingFacets,
    }
  )

  const isLazyRenderEnabled = getSettings('vtex.store')
    ?.enableSearchRenderingOptimization

  return (
    <ProductListProvider listName={trackingId}>
      <div className={galleryClasses}>
        {rows.map((rowProducts, index) => (
          <GalleryRow
            key={index.toString()}
            widthAvailable={width != null}
            products={rowProducts}
            lazyRender={isLazyRenderEnabled && index >= LAZY_RENDER_THRESHOLD}
            summary={summary}
            displayMode={layoutMode}
            rowIndex={index}
            itemsPerRow={itemsPerRow}
            preloadFirstImage={index === 0}
          />
        ))}
        {typeof lazyItemsRemaining === 'number' && lazyItemsRemaining > 0 && (
          <div
            style={{
              width: '100%',
              // Approximate number, just to add scroll leeway
              height: 300 * Math.ceil(lazyItemsRemaining / itemsPerRow),
            }}
            className="flex justify-center pt10"
          >
            <Spinner />
          </div>
        )}
      </div>
      <ProductListEventCaller />
    </ProductListProvider>
  )
}

Gallery.propTypes = {
  /** Container width */
  width: PropTypes.number,
  /** Products to be displayed. */
  products: PropTypes.arrayOf(productShape),
  /** ProductSummary props. */
  summary: PropTypes.any,
  /** Max Items per Row */
  maxItemsPerRow: PropTypes.number,
  /** Layout mode of the gallery in mobile view */
  mobileLayoutMode: PropTypes.oneOf(pluck('value', LAYOUT_MODE)),
  /** Min Item Width. */
  minItemWidth: PropTypes.number,
  showingFacets: PropTypes.bool,
  trackingId: PropTypes.string,
  lazyItemsRemaining: PropTypes.number,
}

export default withResizeDetector(Gallery)
