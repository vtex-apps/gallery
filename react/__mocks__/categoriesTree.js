export default [
  {
    name: 'Eletrônicos',
    link: '/Eletronicos',
    quantity: 0,
    children: [
      {
        name: 'Smartphones',
        link: '/Eletronicos/Smartphones',
        quantity: 0,
        children: [],
      },
      {
        name: 'Videogames',
        link: '/Eletronicos/Videogames',
        quantity: 0,
        children: [],
      },
      {
        name: 'TVs',
        link: '/Eletronicos/TVs',
        quantity: 0,
        children: [],
      },
    ],
  },
  {
    id: 5,
    name: 'Livros',
    link: '/Livros',
    quantity: 0,
    children: [
      {
        id: 6,
        name: 'HQs e Mangas',
        link: '/Livros/HQs e Mangas',
        quantity: 0,
        children: [],
      },
    ],
  },
  {
    id: 7,
    name: 'Roupas',
    link: '/Roupas',
    quantity: 0,
    children: [
      {
        id: 8,
        name: 'Blusas',
        link: '/Roupas/Blusas',
        quantity: 0,
        children: [],
      },
      {
        id: 9,
        name: 'Calças',
        link: '/Roupas/Calcas',
        quantity: 0,
        children: [],
      },
      {
        id: 10,
        name: 'Moletom',
        link: '/Roupas/Moletom',
        quantity: 0,
        children: [],
      },
    ],
  },
]

export const numberOfFilters = 10
