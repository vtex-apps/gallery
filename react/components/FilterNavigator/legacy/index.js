/**
 * WARNING THIS COMPONENT IS DEPRECATED, PLEASE USE THE BLOCK "filter-navigator.v2". THIS COMPONENT WILL BE DELETED IN THE NEXT MAJOR.
 */

import classNames from 'classnames'
import PropTypes from 'prop-types'
import { map, flatten, filter, prop } from 'ramda'
import React, { useMemo } from 'react'
import ContentLoader from 'react-content-loader'
import { FormattedMessage } from 'react-intl'
import { useDevice } from 'vtex.device-detector'

import FilterSidebar from './FilterSidebar'
import SelectedFilters from './SelectedFilters'
import AvailableFilters from './AvailableFilters'
import {
  facetOptionShape,
  paramShape,
  hiddenFacetsSchema,
} from '../../../constants/propTypes'
import useSelectedFilters from './hooks/useSelectedFilters'

import searchResult from './searchResult.css'
import getFilters from './utils/getFilters'

const getCategories = (tree = []) => {
  return [
    ...tree,
    ...flatten(tree.map(node => node.children && getCategories(node.children))),
  ].filter(Boolean)
}

/**
 * Wrapper around the filters (selected and available) as well
 * as the popup filters that appear on mobile devices
 */
const FilterNavigator = ({
  showFilters,
  priceRange,
  tree = [],
  specificationFilters = [],
  priceRanges = [],
  brands = [],
  loading = false,
  hiddenFacets = {},
  preventRouteChange = false,
}) => {
  const { isMobile } = useDevice()

  const filters = getFilters({
    tree,
    specificationFilters,
    priceRanges,
    brands,
    hiddenFacets,
  })
  const selectedFilters = useSelectedFilters(
    useMemo(() => {
      const availableCategories = filter(prop('selected'), getCategories(tree))

      const options = [
        ...availableCategories,
        ...map(prop('facets'), specificationFilters),
        ...brands,
        ...priceRanges,
      ]

      return flatten(options)
    }, [brands, priceRanges, specificationFilters, tree])
  ).filter(facet => facet.selected)

  const filterClasses = classNames({
    'flex justify-center flex-auto bl br b--muted-5': isMobile,
  })

  if (!showFilters) {
    return null
  }

  if (loading && !isMobile) {
    return (
      <div className={searchResult.filters}>
        <ContentLoader
          style={{
            width: '100%',
            height: '100%',
          }}
          width="230"
          height="320"
          y="0"
          x="0"
        >
          <rect width="100%" height="1em" />
          <rect width="100%" height="8em" y="1.5em" />
          <rect width="100%" height="1em" y="10.5em" />
          <rect width="100%" height="8em" y="12em" />
        </ContentLoader>
      </div>
    )
  }

  if (isMobile) {
    return (
      <div className={searchResult.filters}>
        <div className={filterClasses}>
          <FilterSidebar
            filters={filters}
            preventRouteChange={preventRouteChange}
          />
        </div>
      </div>
    )
  }

  return (
    <div className={searchResult.filters}>
      <div className={filterClasses}>
        <div className="bb b--muted-4">
          <h5 className="t-heading-5 mv5">
            <FormattedMessage id="store/search-result.filter-button.title" />
          </h5>
        </div>
        <SelectedFilters
          filters={selectedFilters}
          preventRouteChange={preventRouteChange}
        />
        <AvailableFilters
          filters={filters}
          priceRange={priceRange}
          preventRouteChange={preventRouteChange}
        />
      </div>
    </div>
  )
}

FilterNavigator.propTypes = {
  /** Categories tree */
  tree: PropTypes.arrayOf(facetOptionShape),
  /** Params from pages */
  params: paramShape,
  /** List of brand filters (e.g. Samsung) */
  brands: PropTypes.arrayOf(facetOptionShape),
  /** List of specification filters (e.g. Android 7.0) */
  specificationFilters: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      facets: PropTypes.arrayOf(facetOptionShape),
    })
  ),
  /** List of price ranges filters (e.g. from-0-to-100) */
  priceRanges: PropTypes.arrayOf(facetOptionShape),
  /** Current price range filter query parameter */
  priceRange: PropTypes.string,
  /** Enables or disables filters */
  showFilters: PropTypes.bool,
  /** Loading indicator */
  loading: PropTypes.bool,
  /** Prevents changing route when setting filters (uses URL search params instead) */
  preventRouteChange: PropTypes.bool,
  ...hiddenFacetsSchema,
}

export default FilterNavigator
