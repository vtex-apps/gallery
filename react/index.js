import React, { Component } from 'react'

import SearchResultContainer from './components/SearchResultContainer'
import { SORT_OPTIONS } from './OrderBy'
import LocalQuery from './components/LocalQuery'
import { LAYOUT_MODE } from './components/LayoutModeSwitcher'
import QueryContext from './components/QueryContext'
import SettingsContext from './components/SettingsContext'

const PAGINATION_TYPES = ['show-more', 'infinite-scroll']
const DEFAULT_MAX_ITEMS_PER_PAGE = 10

/**
 * Search Result Query Loader Component.
 * This Component queries the search if the search-result doesn't receive it already
 */
export default class SearchResultQueryLoader extends Component {
  static defaultProps = {
    orderBy: SORT_OPTIONS[0].value,
    showCategoryPanel: false,
    quantityOfItemsPerRow: 4,
  }

  static uiSchema = {
    maxItemsPerLine: {
      'ui:widget': 'radio',
      'ui:options': {
        inline: true,
      },
    },
  }

  render() {
    const {
      querySchema,
      hiddenFacets,
      showFacetsQuantity,
      pagination,
      mobileLayout,
    } = this.props

    const settings = {
      hiddenFacets,
      showFacetsQuantity,
      pagination,
      mobileLayout,
    }

    return !this.props.searchQuery ||
      (querySchema && querySchema.enableCustomQuery) ? (
      <LocalQuery
        {...this.props}
        {...querySchema}
        render={props => (
          <QueryContext.Provider value={props.searchQuery.variables}>
            <SettingsContext.Provider value={settings}>
              <SearchResultContainer {...props} />
            </SettingsContext.Provider>
          </QueryContext.Provider>
        )}
      />
    ) : (
      <QueryContext.Provider value={this.props.searchQuery.variables}>
        <SettingsContext.Provider value={settings}>
          <SearchResultContainer {...this.props} />
        </SettingsContext.Provider>
      </QueryContext.Provider>
    )
  }
}

SearchResultQueryLoader.getSchema = props => {
  const querySchema = !props.searchQuery
    ? {
        querySchema: {
          title: 'admin/editor.search-result.query',
          description: 'admin/editor.search-result.query.description',
          type: 'object',
          properties: {
            maxItemsPerPage: {
              title: 'admin/editor.search-result.query.maxItemsPerPage',
              type: 'number',
              default: DEFAULT_MAX_ITEMS_PER_PAGE,
            },
            queryField: {
              title: 'Query',
              type: 'string',
            },
            mapField: {
              title: 'Map',
              type: 'string',
            },
            orderByField: {
              title: 'Order by field',
              type: 'string',
              default: SORT_OPTIONS[1].value,
              enum: SORT_OPTIONS.map(opt => opt.value),
              enumNames: SORT_OPTIONS.map(opt => opt.label),
            },
            hideUnavailableItems: {
              title: 'admin/editor.search-result.query.hideUnavailableItems',
              type: 'boolean',
              default: false,
            },
          },
        },
      }
    : {}

  return {
    title: 'admin/editor.search-result.title',
    description: 'admin/editor.search-result.description',
    type: 'object',
    properties: {
      ...querySchema,
      mobileLayout: {
        title: 'admin/editor.search-result.mobileLayout',
        type: 'object',
        isLayout: true,
        properties: {
          mode1: {
            title: 'admin/editor.search-result.mobileLayout.mode1',
            type: 'string',
            default: LAYOUT_MODE[0].value,
            enum: LAYOUT_MODE.map(opt => opt.value),
            enumNames: LAYOUT_MODE.map(opt => opt.label),
            isLayout: true,
          },
          mode2: {
            title: 'admin/editor.search-result.mobileLayout.mode2',
            type: 'string',
            default: LAYOUT_MODE[1].value,
            enum: LAYOUT_MODE.map(opt => opt.value),
            enumNames: LAYOUT_MODE.map(opt => opt.label),
            isLayout: true,
          },
        },
      },
      hiddenFacets: {
        title: 'admin/editor.search-result.hiddenFacets',
        type: 'object',
        isLayout: true,
        properties: {
          brands: {
            title: 'admin/editor.search-result.hiddenFacets.brands',
            type: 'boolean',
            isLayout: true,
          },
          categories: {
            title: 'admin/editor.search-result.hiddenFacets.categories',
            type: 'boolean',
            isLayout: true,
          },
          priceRange: {
            title: 'admin/editor.search-result.hiddenFacets.priceRange',
            type: 'boolean',
            isLayout: true,
          },
          specificationFilters: {
            title:
              'admin/editor.search-result.hiddenFacets.specificationFilters',
            type: 'object',
            isLayout: true,
            properties: {
              hideAll: {
                title:
                  'admin/editor.search-result.hiddenFacets.specificationFilters.hideAll',
                type: 'boolean',
                isLayout: true,
              },
              hiddenFilters: {
                type: 'array',
                isLayout: true,
                items: {
                  title:
                    'admin/editor.search-result.hiddenFacets.specificationFilters.hiddenFilter',
                  type: 'object',
                  isLayout: true,
                  properties: {
                    name: {
                      title:
                        'admin/editor.search-result.hiddenFacets.specificationFilters.hiddenFilter.name',
                      type: 'string',
                      isLayout: true,
                    },
                  },
                },
              },
            },
          },
        },
      },
      pagination: {
        type: 'string',
        title: 'admin/editor.search-result.pagination.title',
        default: 'infinity-scroll',
        enum: PAGINATION_TYPES,
        enumNames: [
          'admin/editor.search-result.pagination.show-more',
          'admin/editor.search-result.pagination.infinite-scroll',
        ],
      },
      showCategoryPanel: {
        title: 'editor.search-result.show-category-panel.title',
        type: 'boolean',
        default: false,
        isLayout: true,
      },
      quantityOfItemsPerRow: {
        type: 'number',
        title: 'editor.search-result.quantity-of-items-per-row.title',
        enum: [2, 4],
        default: 4,
        widget: {
          'ui:widget': 'radio',
          'ui:options': {
            inline: true,
          },
        },
        isLayout: true,
      },
      showFacetQuantity: {
        type: 'boolean',
        title: 'admin/editor.search-result.facet-quantity',
        default: false,
      },
    },
  }
}
