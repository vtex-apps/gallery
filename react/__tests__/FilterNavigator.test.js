/* eslint-env jest */
import React from 'react'
import { render } from '@vtex/test-tools/react'

import categoriesTree from 'categoriesTree'

import FilterNavigator from '../FilterNavigator'

describe('<FilterNavigator />', () => {
  const renderComponent = customProps => {
    const props = {
      map: 'c',
      rest: '',
      tree: categoriesTree,
      ...customProps,
    }

    return render(<FilterNavigator {...props} />)
  }

  it('should match snapshot with all', () => {
    const { asFragment } = renderComponent()

    expect(asFragment()).toMatchSnapshot()
  })

  it('should display related categories', () => {
    const { asFragment } = renderComponent({ query: 'Livros' })

    expect(asFragment()).toMatchSnapshot()
  })
})
