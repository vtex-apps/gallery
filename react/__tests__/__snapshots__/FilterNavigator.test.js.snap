// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterNavigator /> should display related categories 1`] = `<DocumentFragment />`;

exports[`<FilterNavigator /> should match snapshot with all 1`] = `<DocumentFragment />`;
